iohk-monitoring/src/Cardano/BM/Data/MonitoringEval.lhs

```diff
@@ -112,8 +112,25 @@ instance Show MEvExpr where
 If evaluation of a monitoring expression is |True|, then a set of actions are
 executed for alerting.
 \begin{code}
-type MEvAction = Text
+data MEvAction = CreateMessage Severity Text
+               | SetGlobalMinimalSeverity Severity
+               | AlterSeverity LoggerName Severity
+               deriving (Eq)
+
+instance FromJSON MEvAction where
+    parseJSON (String s) =
+        case parseEitherAction s of
+            Left e     -> error e
+            Right expr -> pure expr
+    parseJSON _ = error "cannot parse such an action!"
```

Comment:
Use `typeMismatch` from `Data.Aeson` here as
` parseJSON o -> typeMismatch "String" o`

Reply:
the next PR will bring in `fail` and `typeMismatch`. no need to change it now.